test_project/tests/test_adam.py

```diff
@@ -74,3 +74,6 @@ def test_trigger_infinite_loop(self):
 
     def test_single_iteration(self):
         self.assertTrue(adam.single_iteration())
+
+    def test_raise_exception(self):
```

Comment:
Related to an earlier comment: I think this should just assert that the right exception is thrown.